src/memory.h

```diff
@@ -125,20 +102,11 @@ extern void *mem_realloc_func(void *old_ptr, size_t size
 #define str_alloc_copy(a) str_alloc_copy_func(a)
 #define mem_alloc_align(a,b) mem_alloc_align_func(a,b)
 #define mem_calloc_align(a,b,c) mem_calloc_align_func(a,b,c)
-#endif
```

Comment:
All these are memdbg stuff. The functions should be refactored to their original names (eg. mem_alloc_func -> mem_alloc) and the macros should be dropped.

Afterwards, a `git diff master memory.[ch]` should show the Jumbo add-ons with no hoops.

Reply:
good catch, I forgot why these were added.  This is why I was wanting as much review as possible.  The removal was not hard at all, BUT there were quite a few places it was dug in a bit, and not as obvious to be memdbg code

Reply:
I think you will like memory.[ch] differences against master now.

I do think we should move endianity into it's own c and h file (same for the mem-dumping stuff).  Easily included from within the memory.h file (so we have no loss of functionality).   That way, memory.[ch] really does stay close to master, only leaving it for things like small allocation cleanup, addition of realloc/calloc, etc.  But it is all memory allocation/free stuff, and the other ancillary functionality is still declared through a include of memory.h (like it is today), but the baggage needed in memory.[ch] is simply 2 include lines in memory.h (jumbo version).

Just my $.02 worth.   Not sure if that change 'belongs' here, since it has nothing to do with memdbg (so I will not make it now), but once this is done, we may want to think about that one.  IIRC, memory,[ch] is one of the nasty ones for you on merging in core changes.  Now there are simply some error printf that have -+ and the rest are all simply + insertions which again, IIRC, merge like a champ.

Reply:
> I think you will like memory.[ch] differences against master now.

I do! Very clean now.

> I do think we should move endianity into it's own c and h file (same for the mem-dumping stuff). Easily included from within the memory.h file (so we have no loss of functionality). That way, memory.[ch] really does stay close to master, only leaving it for things like small allocation cleanup, addition of realloc/calloc, etc. But it is all memory allocation/free stuff, and the other ancillary functionality is still declared through a include of memory.h (like it is today), but the baggage needed in memory.[ch] is simply 2 include lines in memory.h (jumbo version).

I think we should post-pone this to after next release. And there's no need to get anal about being close to master, it's just very handy to make have a diff that makes sense. Let's concentrate on release blockers now! For example, your dynamic compiler format is a true gem, it's just that it's not working fully 🤣